LEGO1/omni/src/video/mxdisplaysurface.cpp

```diff
@@ -399,86 +399,78 @@ void MxDisplaySurface::VTable0x28(
 		return;
 	}
 
 	MxU8* data = p_bitmap->GetStart(p_left, p_top);
 
 	MxS32 bytesPerPixel = m_surfaceDesc.ddpfPixelFormat.dwRGBBitCount / 8;
-	if (m_videoParam.Flags().GetF1bit3()) {
-		p_bottom *= 2;
-		p_right *= 2;
-
-		MxU8* surface = (MxU8*) ddsd.lpSurface + (bytesPerPixel * p_right) + (p_bottom * ddsd.lPitch);
-		MxLong stride = -p_width + GetAdjustedStride(p_bitmap);
-		MxS32 copyWidth = p_width * bytesPerPixel * 2;
-		MxLong length = -(copyWidth) + ddsd.lPitch;
+	MxU8* surface = (MxU8*) tempDesc.lpSurface;
 
-		while (p_height--) {
-			MxU8* surfaceBefore = surface;
+	MxLong stride = (bytesPerPixel == 1) ? GetAdjustedStride(p_bitmap) : -p_width + GetAdjustedStride(p_bitmap);
+	MxLong length = tempDesc.lPitch - (p_width * bytesPerPixel);
 
-			for (MxS32 i = 0; i < p_width; i++) {
-				if (bytesPerPixel == 1) {
-					surface[0] = surface[1] = *data;
-				}
-				else if (bytesPerPixel == 2) {
-					((MxU16*) surface)[0] = ((MxU16*) surface)[1] = m_16bitPal[*data];
-				}
-				else {
-					((MxU32*) surface)[0] = ((MxU32*) surface)[1] = m_32bitPal[*data];
-				}
-				surface += bytesPerPixel * 2;
-				data++;
+	for (MxS32 i = 0; i < p_height; i++) {
+		if (bytesPerPixel == 1) {
+			memcpy(surface, data, p_width);
+			surface += length + p_width;
+		}
+		else if (bytesPerPixel == 2) {
+			for (MxS32 j = 0; j < p_width; j++) {
+				*(MxU16*) surface = m_16bitPal[*data++];
+				surface += bytesPerPixel;
 			}
 
-			if (stride || length != ddsd.lPitch - copyWidth) {
-				data += stride;
-				surface += length;
+			surface += length;
+		}
+		else {
+			for (MxS32 j = 0; j < p_width; j++) {
+				*(MxU32*) surface = m_32bitPal[*data++];
+				surface += bytesPerPixel;
 			}
-
-			memcpy(surface, surfaceBefore, copyWidth);
-			surface += ddsd.lPitch;
+			surface += length;
 		}
-	}
-	else {
-		MxU8* surface = (MxU8*) ddsd.lpSurface + (bytesPerPixel * p_right) + (p_bottom * ddsd.lPitch);
-		MxLong stride = (bytesPerPixel == 1) ? GetAdjustedStride(p_bitmap) : -p_width + GetAdjustedStride(p_bitmap);
-		MxLong length = ddsd.lPitch - (p_width * bytesPerPixel);
 
-		for (MxS32 i = 0; i < p_height; i++) {
-			if (bytesPerPixel == 1) {
-				memcpy(surface, data, p_width);
-				surface += length + p_width;
-			}
-			else if (bytesPerPixel == 2) {
-				for (MxS32 j = 0; j < p_width; j++) {
-					*(MxU16*) surface = m_16bitPal[*data++];
-					surface += bytesPerPixel;
-				}
+		data += stride;
+	}
 
-				surface += length;
-			}
-			else {
-				for (MxS32 j = 0; j < p_width; j++) {
-					*(MxU32*) surface = m_32bitPal[*data++];
-					surface += bytesPerPixel;
-				}
-				surface += length;
-			}
+	tempSurface->Unlock(NULL);
 
-			data += stride;
-		}
+	RECT destRect;
+	if (m_videoParam.Flags().GetF1bit3()) {
+		destRect = {p_right, p_bottom, p_right + p_width * 2, p_bottom + p_height * 2};
 	}
+	else {
+		destRect = {p_right, p_bottom, p_right + p_width, p_bottom + p_height};
+	}
+	m_ddSurface2->Blt(&destRect, tempSurface, NULL, DDBLT_WAIT, NULL);
```

Comment:
@foxtacles my only idea is try and use `m_ddSurface1`, could you give that a quick test?

Reply:
https://github.com/user-attachments/assets/c3827568-422f-4793-8dac-c629354825a9

That produces a different set of issues: no intro screen, no book wiggling

Reply:
Thanks, it's sort of what I expected, a bit of a long shot really.